httpclient5-fluent/src/main/java/org/apache/hc/client5/http/fluent/ContentResponseHandler.java

```diff
@@ -44,11 +46,63 @@
  */
 public class ContentResponseHandler extends AbstractHttpClientResponseHandler<Content> {
 
+    /**
+     * Handles the response entity and transforms it into a {@link Content} object. If the response entity is null or
+     * has zero length, {@link Content#NO_CONTENT} is returned.
+     *
+     * @param entity the response entity.
+     * @return a {@link Content} object that encapsulates the response body, or {@link Content#NO_CONTENT} if the
+     * response body is null or has zero length.
+     * @throws IOException if an I/O error occurs.
+     */
     @Override
     public Content handleEntity(final HttpEntity entity) throws IOException {
-        return entity != null ?
-                new Content(EntityUtils.toByteArray(entity), ContentType.parse(entity.getContentType())) :
-                Content.NO_CONTENT;
+        if (entity != null && entity.getContentLength() > 0) {
```

Comment:
@arturobernalg Is this reasonable? It is legal for an entity to have its content length as -1 if it is chunk coded.